src/booster.rs

```diff
@@ -8,14 +8,32 @@ use lightgbm_sys;
 
 use crate::{Dataset, Error, Result};
 
+pub const PREDICT_TYPE_NORMAL: i32 = lightgbm_sys::C_API_PREDICT_NORMAL as i32;
+pub const PREDICT_TYPE_LEAF_INDEX: i32 = lightgbm_sys::C_API_PREDICT_LEAF_INDEX as i32;
+
+
+fn load_pandas_categorical(model_string: &str) -> Option<Vec<Vec<Value>>> {
+    let pandas_key = "pandas_categorical:";
```

Comment:
To confirm, this is a standard LGBM way of dumping this information? `pandas_categorical:...` at the end of the model?

Reply:
I guess so, they do it in that way in python implementation https://github.com/microsoft/LightGBM/blob/90342e929b1801fb2a20b410e2db5a69a9063b48/python-package/lightgbm/basic.py#L579

They implemented it only for python, so I ported that